### PR TITLE
Lists cleanup

### PR DESCRIPTION
**Pull request by hypeJunction**
Moves pagination-specific variable out of the lists/gallery components
Cleans up list HTML rendering to take advantage of elgg_format_element()

Manual tests
- [x] Pagination not shown when not needed
- [x] List / gallery classes are built correctly
- [x] Item classes are built correctly

### CODE REVIEW

File: views/default/navigation/pagination.php [hunk 1]
```diff
@@ -17,13 +17,17 @@
 	return true;
 }
 
+$count = (int) elgg_extract('count', $vars, 0);
+if (!$count) {
```
Comment (mrclay) [inline views/default/navigation/pagination.php]: :+1: though prefer `return;`

File: views/default/page/components/list.php [hunk 2]
```diff
@@ -40,60 +36,56 @@
...
-		} else {
-			$id = "item-{$item->getType()}-{$item->id}";
+	$item_view = elgg_view_list_item($item, $vars);
+	if (!$item_view) {
+		continue;
+	}
+
+	$item_classes = $item_class;
+
+	if (elgg_instanceof($item)) {
```
Comment (mrclay) [inline views/default/page/components/list.php]: `$item instanceof ElggEntity`

**Comment by mrclay** [comment]
Love it!

**Comment by hypeJunction** [comment]
@mrclay , your comments are in

**Comment by mrclay** [comment]
LGTM, though we should manually test a bit.

**Comment by hypeJunction** [comment]
Anyone with a vested interest in lists volunteering to do some manual testing? I think someone else doing it would be a more reliable result.

**Comment by hypeJunction** [comment]
This has been rebased, and I have done some more manual testing. This is good for merge, once we figure out what's going on wrong with Travis.

**Comment by hypeJunction** [comment]
Travis is passing. @mrclay perhaps you could take one last look

**Comment by jdalsem** [comment]
LGTM

File: views/default/page/components/gallery.php [hunk 3]
```diff
@@ -42,48 +38,45 @@
 
 elgg_push_context('gallery');
 
-$gallery_class = 'elgg-gallery';
+$gallery_class = array('elgg-gallery');
```
Comment (mrclay) [inline views/default/page/components/gallery.php]: bugs me that these aren't plural var names

File: views/default/page/components/gallery.php [hunk 4]
```diff
@@ -42,48 +38,45 @@
...
+		continue;
+	}
+
+	if ($item instanceof \ElggEntity) {
+		$id = "elgg-{$item->getType()}-{$item->getGUID()}";
+	} else if (is_callable(array($item, 'getType'))) {
+		$id = "item-{$item->getType()}-{$item->id}";
+	}
+
+	$list_items .= elgg_format_element('li', array(
```
Comment (mrclay) [inline views/default/page/components/gallery.php]: Using an $li_attrs would make this easier to read

File: views/default/page/components/gallery.php [hunk 5]
```diff
@@ -42,48 +38,45 @@
...
-			} else {
-				$id = "item-{$item->getType()}-{$item->id}";
-			}
-			echo "<li id=\"$id\" class=\"$item_class\">";
-			echo elgg_view_list_item($item, $vars);
-			echo "</li>";
-		}
-	?>
-</ul>
+echo elgg_format_element('ul', array(
```
Comment (mrclay) [inline views/default/page/components/gallery.php]: `echo elgg_format_element('ul', ['class' => $gallery_class], $list_items);`

**Comment by mrclay** [comment]
Besides my nitpicking, LGTM

**Comment by hypeJunction** [comment]
@mrclay See if it's more readable this way

**Comment by mrclay** [comment]
:clap: :tada:

**Pull request merged**